File: csharp/exercises/Webshop.Service/Common/Logger.cs
namespace Webshop.Service.Common;

public static class Logger
{
    public static void LogError(string message)
    {
        Log("Error", message);
    }

    public static void LogWarning(string message)
    {
        Log("Warning", message);
    }

    public static void LogInfo(string message)
    {
        Log("Info", message);
    }

    private static void Log(string level, string message)
    {
        using (var writer = File.AppendText("\\\\someserversomewhere\\log.txt"))
        {
            writer.WriteLine($"[{level}] [{DateTime.UtcNow.ToString("s")}] {message}");
        }
    }
}

File: java/exercises/src/main/java/Webshop/Service/Common/Logger.java
package Webshop.Service.Common;

import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class Logger {

    public static void logError(String message) {
        log("Error", message);
    }

    public static void logWarning(String message) {
        log("Warning", message);
    }

    public static void logInfo(String message) {
        log("Info", message);
    }

    private static void log(String level, String message) {
        String logFilePath = "\\\\someserversomewhere\\log.txt";
        try (PrintWriter writer = new PrintWriter(new FileWriter(logFilePath, true))) {
            String formattedDateTime = LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
            writer.println("[" + level + "] [" + formattedDateTime + "] " + message);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
